### PR TITLE
Use a lockfile to serialize blessclient invocations

### PR DESCRIPTION
**Pull request by edulop91**
When integrated with tools like terraform or ansible, multiple concurrent runs of blessclient resulted in unnecessary work. Instead, serialize blessclient commands.

### CODE REVIEW

File: pkg/util/cobra.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package util
+
+import (
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// GetConfigPath gets the config path from a cobra cmd
+func GetConfigPath(cmd *cobra.Command) (string, error) {
```
Comment (ryanking) [inline pkg/util/cobra.go]: Probably fine for now but we should maybe consider using [viper](https://github.com/spf13/viper) for this.

File: pkg/util/pid_lock.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			logrus.WithError(err).Warn("Error acquiring lock")
+			return err
+		}
+		return nil
+	}
+	return backoff.Retry(lockFun, l.backoff)
+}
+
+// Unlock will unlock the pid lockfile
+func (l *Lock) Unlock() error {
```
Comment (ryanking) [inline pkg/util/pid_lock.go]: seems fishy

File: pkg/util/cobra.go (hunk 1 above)
Comment (edulop91) [reply pkg/util/cobra.go]: yeah, i have another pr locally where i'm doing this

File: pkg/util/pid_lock.go (hunk 2 above)
Comment (edulop91) [reply pkg/util/pid_lock.go]: heh, yeah

**Pull request merged**